internal/services/mssql/mssql_job_target_group_resource.go

```diff
@@ -0,0 +1,327 @@
+			Required:     true,
+			ValidateFunc: jobs.ValidateJobAgentID,
+			ForceNew:     true,
+		},
+		"job_target": {
+			Type:     pluginsdk.TypeSet,
+			Optional: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"server_name": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validate.ValidateMsSqlServerName,
+					},
+					"type": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+						ValidateFunc: validation.StringInSlice([]string{
+							string(jobtargetgroups.JobTargetTypeSqlDatabase),
+							string(jobtargetgroups.JobTargetTypeSqlElasticPool),
+							string(jobtargetgroups.JobTargetTypeSqlServer),
+						}, false),
+					},
+					"database_name": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ValidateFunc: validate.ValidateMsSqlDatabaseName,
+					},
+					"elastic_pool_name": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ValidateFunc: validate.ValidateMsSqlElasticPoolName,
+					},
+					"job_credential_id": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ValidateFunc: jobcredentials.ValidateCredentialID,
+					},
+					"membership_type": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						Default:      string(jobtargetgroups.JobTargetGroupMembershipTypeInclude),
+						ValidateFunc: validation.StringInSlice(jobtargetgroups.PossibleValuesForJobTargetGroupMembershipType(), false),
+					},
+				},
+			},
+		},
+	}
+}
+
+func (r MsSqlJobTargetGroupResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r MsSqlJobTargetGroupResource) ModelObject() interface{} {
+	return &MsSqlJobTargetGroupResourceModel{}
+}
+
+func (r MsSqlJobTargetGroupResource) ResourceType() string {
+	return "azurerm_mssql_job_target_group"
+}
+
+func (r MsSqlJobTargetGroupResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.MSSQL.JobTargetGroupsClient
+
+			var model MsSqlJobTargetGroupResourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return err
+			}
+
+			jobAgent, err := jobtargetgroups.ParseJobAgentID(model.JobAgentID)
+			if err != nil {
+				return err
+			}
+
+			id := jobtargetgroups.NewTargetGroupID(jobAgent.SubscriptionId, jobAgent.ResourceGroupName, jobAgent.ServerName, jobAgent.JobAgentName, model.Name)
+
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			targets, err := expandJobTargets(model.JobTargets)
+			if err != nil {
+				return err
```

Comment:
Errors returned by expanding/flattening functions are usually wrapped with the block's name in backticks
```suggestion
				return fmt.Errorf("expanding `job_target`: %+v", err)
```